editor/src/clj/editor/atlas.clj

```diff
@@ -1109,3 +1098,156 @@
+                     (conj 0.0))]
+    (concat
+     (scene-tools/vtx-add position (scene-tools/vtx-scale [7.0 7.0 1.0] (scene-tools/gen-circle 64)))
+     (scene-tools/vtx-add position (scene-tools/gen-point)))))
+
+(g/defnk produce-manip-delta
+  [original-values manip-space start-action action]
+  (when (and start-action action)
+    (let [{:keys [world-rotation world-transform]} (peek original-values)
+          manip-origin (math/translation world-transform)
+          manip-rotation (scene-tools/get-manip-rotation manip-space world-rotation)
+          lead-transform (doto (Matrix4d.) (.set manip-origin))
+          [start-pos pos] (map #(let [manip-dir (math/rotate manip-rotation (Vector3d. 0.0 0.0 1.0))
+                                      _ (.transform lead-transform manip-dir)
+                                      _ (.normalize manip-dir)
+                                      manip-pos (math/translation lead-transform)]
+                                  (math/line-plane-intersection (:world-pos %) (:world-dir %) (Point3d. manip-pos) manip-dir)) [start-action action])]
+      (doto (Vector3d.) (.sub pos start-pos)))))
+
+(g/defnk produce-renderables [_node-id manip-space camera viewport selected-renderables manip-delta snap-enabled snap-threshold]
+  (if (or (empty? selected-renderables)
+          (second selected-renderables)
+          (not= (-> selected-renderables last :node-id g/node-by-id g/node-type) AtlasImage))
+    {}
+    (let [reference-renderable (last selected-renderables)
+          {:keys [node-id world-translation world-rotation]} reference-renderable
+          scale (scene-tools/scale-factor camera viewport world-translation)
+          world-transform (scene-tools/manip-world-transform reference-renderable manip-space scale)
+          pivot-pos (if manip-delta
+                      (cond-> (updated-pivot node-id manip-delta)
+                        snap-enabled (snap-pivot snap-threshold))
+                      (g/node-value node-id :pivot))
+          vertices (pivot-handle reference-renderable pivot-pos scale)
+          inv-view (doto (c/camera-view-matrix camera) (.invert))
+          renderable (scene-tools/gen-manip-renderable _node-id :move-pivot manip-space world-rotation world-transform (AxisAngle4d.) vertices colors/defold-turquoise inv-view)]
+      {pass/manipulator [renderable]
+       pass/manipulator-selection [renderable]})))
+
+(g/defnk produce-scale [original-values camera viewport]
+  (when-not (empty? original-values)
+    (let [{:keys [world-transform]} (peek original-values)
+          manip-origin (math/translation world-transform)
+          lead-transform (doto (Matrix4d.) (.set manip-origin))
+          manip-pos (math/translation lead-transform)]
+      (scene-tools/scale-factor camera viewport manip-pos))))
+
+(defn handle-input [self action selection-data]
+  (case (:type action)
+    :mouse-pressed (if (first (get selection-data self))
```

Comment:
Maybe here, on mouse pressed, we could inspect the selection, filter the atlas images, and then connect the pivots to the controller?